src/main/java/org/gwtproject/activity/widget/shared/WidgetActivityManager.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2010 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.gwtproject.activity.widget.shared;
+
+import org.gwtproject.activity.shared.ActivityDisplay;
+import org.gwtproject.activity.shared.ActivityManager;
+import org.gwtproject.event.shared.EventBus;
+import org.gwtproject.user.client.ui.AcceptsOneWidget;
+import org.gwtproject.user.client.ui.IsWidget;
+
+/**
+ * Drop-in replacement for {@link com.google.gwt.activity.shared.ActivityManager}.
+ */
+public class WidgetActivityManager extends ActivityManager<IsWidget> {
+
+  private AcceptsOneWidget widgetDisplay;
```

Comment:
> Looks like it's never assigned, maybe it doesn't really matter if passing the same display should be a no-op or not (ActivityManager doesn't really care, all it cares about apparently is whether we go from null to non-null, or from non-null to null), but it's a breaking change.

Indeed, I didn't see that. I saw it was comparing it (instance compare), so I took the safe route and decided to provide same wrapper for same delegate. If my 2 cents are worth anything, I wouldn't care about instancing new wrapper for each call since it results in much more readable code.

> This should be assigned in setDisplay, and tested for non-regression wrt c.g.g.activity.

Yes, that was intention, but I forgot and since `ActivityManager` really doesn't care (like you pointed), it didn't matter in tests.

I'll initialize it, but will leave a note for possible simplification (with performance penalty, though).